components/script/dom/customelementregistry.rs

```diff
@@ -95,7 +96,15 @@ impl CustomElementRegistryMethods for CustomElementRegistry {
         rooted!(in(global_scope.get_cx()) let constructor = constructor_.callback());
 
         // Step 1
-        if unsafe { !IsConstructor(constructor.get()) } {
+        // We must unwrap the constructor as all wrappers are constructable if they are callable.
+        rooted!(in(global_scope.get_cx()) let unwrapped_constructor = unsafe { UnwrapObject(constructor.get(), 1) });
```

Comment:
Should I be storing the unwrapped constructor in `CustomElementDefinition` instead of `Rc<Function>`? Or do we need to store both?